presto-orc/src/main/java/com/facebook/presto/orc/writer/TimestampColumnWriter.java

```diff
@@ -135,7 +136,7 @@ public long writeBlock(Block block)
         int blockNonNullValueCount = 0;
         for (int position = 0; position < block.getPositionCount(); position++) {
             if (!block.isNull(position)) {
-                long value = type.getLong(block, position);
+                long value = microsToMillis(type.getLong(block, position));
```

Comment:
ORC supports storing timestamps with nanosecond precision. Why do we need to truncate to milliseconds here?

Reply:
> ORC supports storing timestamps with nanosecond precision.

At present, only microsecond fake has been made for ORC format. I think we can support real microsecond precision of ORC in later PR.

> Why do we need to truncate to milliseconds here?

Since microsecond precision in ORC format is not really supported yet, this change is made to be compatible with existing logic. The milliseconds will be padded to microsecond with 0 later.